src/v/cloud_storage/topic_manifest.cc

```diff
@@ -323,7 +323,7 @@ void topic_manifest::serialize(std::ostream& out) const {
     json::Writer<json::OStreamWrapper> w(wrapper);
     w.StartObject();
     w.Key("version");
-    w.Int(static_cast<int>(manifest_version::v1));
+    w.Int(static_cast<int>(topic_manifest_version));
```

Comment:
>cloud_storage: fix version in topic_manifest::serialize

it would be useful to leave an extra sentence in the commit message that explains what is being fixed.